libraries/AP_Scripting/drivers/LTE_modem.lua

```diff
@@ -148,49 +195,294 @@ local log_file = io.open('SIM7600.log', 'w')
 --]]
 local function uart_read()
     local s = uart:readstring(512)
-    if s and #s > 0 and log_file then
-        log_file:write('[' .. s .. ']\n')
-        log_file:flush()
-    end
+    log_data(s, '<<<')
+    stats.bytes_in = stats.bytes_in + #s
     return s
 end
 
+--[[
+    Function to write to the UART and log the command
+--]]
+local function uart_write(s)
+    uart:writestring(s)
+    log_data(s, '>>>')
+    stats.bytes_out = stats.bytes_out + #s
+    return #s
+end
+
+-- Constants for GSM 07.10 CMUX framing
+local FLAG = 0xF9
+local UIH = 0xEF
+local SABM = 0x2F
+--local UA = 0x63
+local EA = 0x01
+local CR_SEND = 0x02
+
+-- CMUX buffer state
+local cmux = {}
+cmux.buffers = {[1] = "", [2] = ""} -- DLC1=AT, DLC2=DATA(PPP or TCP)
+
+--[[
+    FCS lookup table for polynomial x^8 + x^2 + x^1 + 1 (0x07)
+    This is the reverse of the standard CRC-8 table
```

Comment:
Do we have this in CPP? I wonder it if would make sense to expose it. (not in this PR tho) .

Reply:
I don't know, but I want the script to work with 4.6.x anyway